okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java

```diff
@@ -951,9 +958,9 @@ void finishStream(
                   stopDelivery,
                   trailers != null ? trailers : new Metadata());
         }
+        maybeClearInUse(stream);
```

Comment:
Hmm... looks like we should really remove the in-use tracker entirely, as we're already keeping track of that stuff already (all the places that check `streams.isEmpty() && pendingStreams.isEmpty()` or similar). But that's a bigger change, so this approach is fine for now.

But we should clear the in-use after the `startPendingStreams()` block to avoid toggling in-use back and forth rapidly.